### PR TITLE
[HIP][UR] Fix memory type detection in allocation info queries and USM copy2D

### PR DESCRIPTION
**Pull request by GeorgeWeb**
Test CI for https://github.com/oneapi-src/unified-runtime/pull/1455

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -4,9 +4,9 @@
 # Options to override the default behaviour of the FetchContent to include UR
 # source code.
 set(SYCL_PI_UR_OVERRIDE_FETCH_CONTENT_REPO
-  "" CACHE STRING "Override the Unified Runtime FetchContent repository")
+  "https://github.com/GeorgeWeb/unified-runtime.git" CACHE STRING "Override the Unified Runtime FetchContent repository")
 set(SYCL_PI_UR_OVERRIDE_FETCH_CONTENT_TAG
-  "" CACHE STRING "Override the Unified Runtime FetchContent tag")
+  "92b60b77959420db8938770319ba92f7fa6853ef" CACHE STRING "Override the Unified Runtime FetchContent tag")
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:9]: We should revert these changes.

**Review by kbenzie — CHANGES_REQUESTED** [review]

**Review by aarongreig — APPROVED** [review]

**Comment by aarongreig** [comment]
@intel/llvm-gatekeepers please merge
